BlitzBoard/Features/FixturesScreenViewController.swift

```diff
@@ -24,6 +25,21 @@ class FixturesScreenViewController: UIViewController {
         viewModel.fetchFixtures()
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        spinner.isHidden = false
+        spinner.startAnimating()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        do {
+            sleep(3)
```

Comment:
so usually you don't want to have this on a timer, you want to link this to when your service call is actually complete. So the spinner.stopAnimating() will be called in the .finished of the viewModel when it fetches the data from the service